src/hotspot/share/gc/z/zVirtualMemory.cpp

```diff
@@ -99,16 +98,56 @@ size_t ZVirtualMemoryManager::reserve_discontiguous(size_t size) {
   return reserved;
 }
 
+bool ZVirtualMemoryManager::reserve_contiguous_inner(uintptr_t start, size_t size) {
+  assert(is_aligned(size, ZGranuleSize), "Must be granule aligned");
+
+  // Reserve address views
+  const uintptr_t marked0 = ZAddress::marked0(start);
+  const uintptr_t marked1 = ZAddress::marked1(start);
+  const uintptr_t remapped = ZAddress::remapped(start);
+
+  // Reserve address space
+  if (!os_reserve(marked0, size)) {
+    return false;
+  }
+
+  if (!os_reserve(marked1, size)) {
+    os_unreserve(marked0, size);
+    return false;
+  }
+
+  if (!os_reserve(remapped, size)) {
+    os_unreserve(marked0, size);
+    os_unreserve(marked1, size);
+    return false;
+  }
+
+  // Register address views with native memory tracker
+  nmt_reserve(marked0, size);
+  nmt_reserve(marked1, size);
+  nmt_reserve(remapped, size);
+
+  return true;
+}
+
+bool ZVirtualMemoryManager::reserve_contiguous(uintptr_t start, size_t size) {
+  if (reserve_contiguous_inner(start, size)) {
+    // Make the address range free
+    _manager.free(start, size);
```

Comment:
I think we can just move these two lines to the end of reserve_contiguous_inner(), and remove this function.

Reply:
Good point.